### PR TITLE
📖   Move all docs to the book

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**: Move all docs to the book to have one central place for documentation.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Move all docs to the book
```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/hold fixing some links

**Comment by CecileRobertMichon** [comment]
https://deploy-preview-1316--kubernetes-sigs-cluster-api-provider-azure.netlify.app/

**Comment by CecileRobertMichon** [comment]
/retest

working on fixing the flake in #1315

File: docs/book/src/topics/cloud-provider-config.md [hunk 1]
```diff
@@ -5,3 +5,31 @@ The [Azure cloud provider](https://github.com/kubernetes-sigs/cloud-provider-azu
...
 
 For AzureMachineTemplate and standalone AzureMachines, the generated secret will have the name "${RESOURCE}-azure-json", where "${RESOURCE}" is the name of either the AzureMachineTemplate or AzureMachine. The secret will have one data field, `azure.json`, with the raw content for that file. When the secret `${RESOURCE}-azure-json` already exists in the same namespace as an AzureCluster and does not have the label `"${CLUSTER_NAME}": "owned"`, CAPZ will not generate the default described above. Instead it will directly use whatever the user provides in that secret.
+
+# External Cloud Provider
+
+To deploy a cluster using [external cloud provider](https://github.com/kubernetes-sigs/cloud-provider-azure), create a cluster configuration with the [external cloud provider template](https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/master/templates/cluster-template-external-cloud-provider.yaml).
+
+After components are deployed, you should see following pods in `Running` state:
```
Comment (mboersma) [inline docs/book/src/topics/cloud-provider-config.md:13]:
```suggestion
After deploying the cluster, you should eventually see a set of pods like the following in a `Running` state:
```

Comment (mboersma) [reply docs/book/src/topics/cloud-provider-config.md:13]: Not sure if we wanted any actual content changes here, so feel free to ignore this--I can submit a separate PR.

**Review by mboersma — APPROVED** [review]
/lgtm

File: docs/book/src/topics/cloud-provider-config.md (hunk 1 above)
Comment (CecileRobertMichon) [reply docs/book/src/topics/cloud-provider-config.md]: applied as a separate commit

**Review by devigned — APPROVED** [review]
/lgtm

**Pull request merged**